compiler/modules/print.llvm/llvm.bal

```diff
@@ -266,20 +266,46 @@ public class FunctionDecl {
     final FunctionType functionType;
     final string functionName;
     string? gcName = ();
-    final EnumAttribute[] attributes = [];
+    final FunctionEnumAttribute[] functionAttributes = [];
+    final ReturnEnumAttribute[] returnAttributes = [];
+    final ParamEnumAttribute[][] paramAttributes = [];
 
     function init(Context context, string functionName, FunctionType functionType) {
         self.functionName = functionName;
         self.functionType = functionType;
+        foreach var _i in 0..< functionType.paramTypes.length() {
+           self.paramAttributes.push([]); 
+        }
     }
 
     function output(Output out) {
         out.push(functionHeader(self));
     }
 
     public function addEnumAttribute(EnumAttribute attribute) {
-        if self.attributes.indexOf(attribute) == () {
-           self.attributes.push(attribute); 
+        if attribute is FunctionEnumAttribute {
+            self.pushAttributeToContainer(self.functionAttributes, attribute);
+        }
+        else {
+            if attribute[0] == "return" {
+                ReturnEnumAttribute attrib = attribute[1];
+                self.pushAttributeToContainer(self.returnAttributes, attrib);
+            } else {
+                ParamEnumAttribute attrib = <ParamEnumAttribute>attribute[1];
+                int paramIndex = <int>attribute[0];
+                if paramIndex >= self.paramAttributes.length() {
+                    panic err:illegalArgument("Invalid index for parameter attribute");
+                } else {
+                    self.pushAttributeToContainer(self.paramAttributes[paramIndex], attrib);
+                }
+            }
+        }
+    }
+
+    function pushAttributeToContainer((FunctionEnumAttribute|ParamEnumAttribute|ReturnEnumAttribute)[] container,
```

Comment:
Not a great function name. You are using the list to represent a set. So maybe `addAttributeToSet`.  I would just use `string` instead of `FunctionEnumAttribute|ParamEnumAttribute|ReturnEnumAttribute`, since it isn't type-safe here (and should be `private`).

Reply:
Fixed